packages/react-component-library/src/components/Timeline/hooks/useTimelinePosition.ts

```diff
@@ -1,28 +1,34 @@
 import { useContext } from 'react'
 
 import {
-  differenceInSeconds,
+  differenceInCalendarDays,
   isBefore,
   isAfter,
 } from 'date-fns'
 
 import { TimelineContext } from '../context'
 import { formatPx } from '../helpers'
 
-const SECONDS_IN_DAY = 86400
+const timeOffset = (date: Date) => 1 / 24 * new Date(date).getHours()
 
-function getWidth(startDate: Date, endDate: Date): number {
-  return (
-    differenceInSeconds(new Date(endDate), new Date(startDate)) /
-    SECONDS_IN_DAY
-  )
+function getWidth (
+  startDate: Date,
+  endDate: Date
+): number {
+  return differenceInCalendarDays(
+    new Date(endDate),
+    new Date(startDate)
+  ) + timeOffset(endDate)
```

Comment:
Do we need to consider the case where the first day is partial one here as well?

Reply:
I'll add some tests but I think this now makes sense.